### ISSUE TITLE
trainer.py: Dead code — loss_per_sample assignment is overwritten before use

### ISSUE THREAD

**Issue by ArgonZapan** [ArgonZapan/DQN-transformer#1]
## Bug Description

In `python/training/trainer.py`, the variable `loss_per_sample` is computed at line 148 but then immediately overwritten in the PER code path at line 151, making the assignment at line 148 unreachable when `use_per=True`.

```python
# trainer.py:146-154
td_errors = targets - current_q_values
loss_per_sample = self.loss_fn(current_q_values, targets)   # line 148

if is_weights is not None:
    loss = (loss_per_sample * is_weights).mean()              # line 151 — overwrites
else:
    loss = loss_per_sample.mean()                             # line 153 — unreachable with PER
```

When `use_per=True`, the value of `loss_per_sample` is multiplied by `is_weights` *inside* the if-block, so the assignment at line 148 is computed but never used in that path. The variable then gets the correct result assigned at line 151, making line 153 dead code for the PER path.

While the *end result* is correct (loss is properly computed and backpropagated), this is misleading during code review and maintenance — the variable name implies it holds per-sample losses ready for weighting, but the actual per-sample losses used are the result of line 151's assignment, not line 148's.

## Expected Behavior

The code should be restructured so that `loss_per_sample` is either:
- Only computed in the non-PER path, **or**
- Computed once and then weighted/averaged consistently in both paths.

## Suggested Fix

Move `self.loss_fn()` inside the `else` block, or restructure to compute loss once and apply `is_weights` consistently:

```python
td_errors = targets - current_q_values

if is_weights is not None:
    per_sample_loss = self.loss_fn(current_q_values, targets)
    loss = (per_sample_loss * is_weights).mean()
else:
    loss = self.loss_fn(current_q_values, targets).mean()
```

## Labels

bug, code-quality
